pkg/reconciler/openshift/tektonaddon/transformer.go

```diff
@@ -133,3 +137,79 @@ func setVersionedNames(operatorVersion string) mf.Transformer {
 		return nil
 	}
 }
+
+// replacePACTriggerTemplateImages replaces images in all the TriggerTemplates that
+// PAC creates. These TriggerTemplates have the "same" PAC image in all
+// TaskRun.Spec.TaskSpec.Steps irrespective of the step (or container) name; this
+// is what makes this transformer a bit different that other replace-image
+// transformers. In other transformers, the images are replaced based on the
+// container name in Deployments, but here the container names in steps are
+// irrelevant because we need to replace any image in "all" steps present with the
+// PAC image.
+func replacePACTriggerTemplateImages(pacImage string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "TriggerTemplate" {
+			return nil
+		}
+
+		tt := &triggersv1beta1.TriggerTemplate{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, tt)
+		if err != nil {
+			return err
+		}
+
+		for i, resourceTemplate := range tt.Spec.ResourceTemplates {
+			// let' start by checking that we are only dealing with a TaskRun
+
+			// first we need to decode the raw ResourceTemplate
+			rawReader := mf.Reader(bytes.NewReader(resourceTemplate.RawExtension.Raw))
+			rawObjects, err := rawReader.Parse()
+			if err != nil {
+				return err
+			}
+
+			if len(rawObjects) == 0 {
```

Comment:
I am not able to think of any case making this

Reply:
me neither, but I'd rather err on the side of caution here 🙃